server/src/Service/InteractionService.php

```diff
@@ -43,17 +43,20 @@ public function getInteraction(
 		$startDate = $this->sanitizeDate( $startDate );
 		$endDate = $this->sanitizeDate( $endDate );
 
-		// get user ids from usernames
-		$userIds = array_map( function ( $username ) {
-			$userId = $this->userDao->getUserId( $username );
-			if ( !$userId ) {
-				throw new \InvalidArgumentException( sprintf( 'username %s could not be found', $username ) );
-			}
-			return $userId;
-		}, $users );
+		// If none of the users is an IP, get their user ID(s) to improve performance
```

Comment:
Just curious if you have done any testing that shows `rev_user` is faster? We recently added IP support to XTools, and for this I simply rewrote most queries to use `rev_user_text` regardless of whether the inputted user was an IP or account, and I didn't notice any difference in performance. When I did EXPLAIN on the queries (on the replicas), I did see slightly fewer estimated rows with `rev_user`, but again even for users with a lot of edits the run time seems to be roughly the same.

Reply:
I remember it was significantly faster while I was working on it. Mostly the query for common pages. I'll run some examples tomorrow and get some proper benchmarks for you